src/material.h

```diff
@@ -75,4 +77,13 @@ class material_type
         const mat_burn_data &burn_data( size_t intensity ) const;
 };
 
+namespace materials
```

Comment:
In any case the nomenclature could be much better: consider material vs material**s**. Using the plural form is going to cause problems especially if we try and update all the other `string_id` instances to use this system. For example what is the plural of sheep?

> I can (will) unify the relevant code within a single (next) PR

> What I'd like to move to is an explicit top-level object (owned by main or the main game loop) that owns all these little factories, then we can stop having bizarre crashes due to static initialization

This should be the aim. Lets design something and have everything use it. The current system is a complete mishmash.